File: src/core/jupiter/core/domain/sync_target.py
"""What exactly to sync."""

from jupiter.core.framework.value import EnumValue, enum_value


@enum_value
class SyncTarget(EnumValue):
    """What exactly to generate, gc, or look at systematically."""

    INBOX_TASKS = "inbox-tasks"
    WORKING_MEM = "working-mem"
    TIME_PLANS = "time-plans"
    SCHEDULES = "schedule"
    HABITS = "habits"
    CHORES = "chores"
    BIG_PLANS = "big-plans"
    JOURNALS = "journals"
    DOCS = "docs"
    VACATIONS = "vacations"
    PROJECTS = "projects"
    SMART_LISTS = "smart-lists"
    METRICS = "metrics"
    PERSONS = "persons"
    SLACK_TASKS = "slack-tasks"
    EMAIL_TASKS = "email-tasks"

File: src/webui/app/logic/domain/sync-target.ts
import { SyncTarget } from "@jupiter/webapi-client";

export function syncTargetName(syncTarget: SyncTarget): string {
  switch (syncTarget) {
    case SyncTarget.INBOX_TASKS:
      return "Inbox Tasks";
    case SyncTarget.WORKING_MEM:
      return "Working Mem";
    case SyncTarget.TIME_PLANS:
      return "Time Plans";
    case SyncTarget.SCHEDULE:
      return "Schedule";
    case SyncTarget.HABITS:
      return "Habits";
    case SyncTarget.CHORES:
      return "Chores";
    case SyncTarget.BIG_PLANS:
      return "Big Plans";
    case SyncTarget.JOURNALS:
      return "Journals";
    case SyncTarget.DOCS:
      return "Docs";
    case SyncTarget.VACATIONS:
      return "Vacations";
    case SyncTarget.PROJECTS:
      return "Projects";
    case SyncTarget.SMART_LISTS:
      return "Smart Lists";
    case SyncTarget.METRICS:
      return "Metrics";
    case SyncTarget.PERSONS:
      return "Persons";
    case SyncTarget.SLACK_TASKS:
      return "Slack Tasks";
    case SyncTarget.EMAIL_TASKS:
      return "Email Tasks";
    case SyncTarget.GAMIFICATION:
      return "Gamification";
  }
}
